maestro-cli/src/main/java/maestro/cli/command/TestCommand.kt

```diff
@@ -119,70 +141,186 @@ class TestCommand : Callable<Int> {
+                                val deviceCreated = DeviceCreateUtil.getOrCreateDevice(
+                                    cfg.platform,
+                                    cfg.osVersion,
+                                    null,
+                                    null,
+                                    true,
+                                    shardIndex
+                                )
+
+                                DeviceService.startDevice(deviceCreated, driverHostPort, initialActiveDevices + currentActiveDevices).instanceId.also {
+                                    currentActiveDevices.add(it)
+                                    delay(10.seconds)
+                                }
+                            }
+                    // Release lock if device ID was obtained from the connected devices
+                    deviceCreationSemaphore.release()
+                    // Signal that this thread has reached the barrier
+                    barrier.countDown()
+                    // Wait for all threads/shards to complete device creation before proceeding
+                    barrier.await()
+
+                    MaestroSessionManager.newSession(
+                        host = parent?.host,
+                        port = parent?.port,
+                        driverHostPort = driverHostPort,
+                        deviceId = deviceId
+                    ) { session ->
+                        val maestro = session.maestro
+                        val device = session.device
+
+                        if (flowFile.isDirectory || format != ReportFormat.NOOP) {
+                            if (continuous) {
+                                throw CommandLine.ParameterException(
+                                    commandSpec.commandLine(),
+                                    "Continuous mode is not supported for directories. $flowFile is a directory",
+                                )
+                            }
+
+                            val suiteResult = TestSuiteInteractor(
+                                maestro = maestro,
+                                device = device,
+                                reporter = ReporterFactory.buildReporter(format, testSuiteName),
+                            ).runTestSuite(
+                                executionPlan = chunkPlans[shardIndex],
+                                env = env,
+                                reportOut = format.fileExtension
+                                    ?.let { extension ->
+                                        (output ?: File(
+                                            "report${
+                                                if (shardIndex >= 1) {
+                                                    "_" + shardIndex + 1
+                                                } else {
+                                                    ""
+                                                }
```

Comment:
As a user I feel having to deal with many junit xml files one for each shard is unexpected. Since maestro is handling the parallel execution. I'd expect that we combine them into a single file, given that they all share the same test suite name.

Reply:
Sure, I'll address this soon, thank you 👍